src/transport/msg_providers/data_bus.py

```diff
@@ -0,0 +1,281 @@
+        deposit_root, nonce, block_number, block_hash, signature, staking_module_id, app = parsed_data
+        r, _vs = signature_to_r_vs(signature)
+        return PauseMessage(
+            type=MessageType.PAUSE,
+            blockNumber=block_number,
+            guardianAddress=guardian,
+            stakingModuleId=staking_module_id,
+            signature={
+                'r': r,
+                '_vs': _vs,
+            },
+        )
+
+
+# event MessagePauseV3(address indexed guardianAddress, (uint256 blockNumber, bytes signature, (bytes32 version) app) data)",
+
+
+class PauseV3Parser(LogParser):
+    def __init__(self, w3: Web3):
+        super().__init__(w3, PAUSE_V3_DATA_SCHEMA)
+
+    def _create_message(self, parsed_data: tuple, guardian: str) -> dict:
+        block_number, signature, app = parsed_data
+        r, _vs = signature_to_r_vs(signature)
+        return PauseMessage(
+            type=MessageType.PAUSE,
+            blockNumber=block_number,
+            guardianAddress=guardian,
+            signature={
+                'r': r,
+                '_vs': _vs,
+            },
+        )
+
+
+class DataBusSinks(str, Enum):
+    DEPOSIT_V1 = f'MessageDepositV1(address,{DEPOSIT_V1_DATA_SCHEMA})'
+    PAUSE_V2 = f'MessagePauseV2(address,{PAUSE_V2_DATA_SCHEMA})'
+    PAUSE_V3 = f'MessagePauseV3(address,{PAUSE_V3_DATA_SCHEMA})'
+    PING_V1 = f'MessagePingV1(address,{PING_V1_DATA_SCHEMA})'
+    UNVET_V1 = f'MessageUnvetV1(address,{UNVET_V1_DATA_SCHEMA})'
+
+
+def _construct_parsers(w3: Web3, sinks: List[DataBusSinks]) -> List[LogParser]:
+    parser_mapping = {
+        DataBusSinks.DEPOSIT_V1: DepositParser,
+        DataBusSinks.PAUSE_V2: PauseV2Parser,
+        DataBusSinks.PAUSE_V3: PauseV3Parser,
+        DataBusSinks.PING_V1: PingParser,
+        DataBusSinks.UNVET_V1: UnvetParser,
+    }
+
+    parsers = []
+    for sink in sinks:
+        parser_class = parser_mapping.get(sink)
+        if parser_class:
+            parsers.append(parser_class(w3))
+        else:
+            raise ValueError(f'Invalid sink in Data Bus sinks: {sink}')
+
+    return parsers
+
+
+class DataBusProvider(BaseMessageProvider):
+    def __init__(self, w3: Web3, message_schema: Schema, sinks: [DataBusSinks]):
+        super().__init__('', message_schema)
+        if len(sinks) == 0:
+            raise ValueError('There must be at least a single sink for Data Bus provider')
+
+        if not sinks:
+            raise ValueError('There must be at least one sink for Data Bus provider')
+
+        self._STANDARD_OFFSET = 256
+        self._latest_block = -1
+        self._queue: deque = deque()
+
+        logger.info('Data bus client initialized.')
+
+        self._w3 = w3
+        self._topics = [self._w3.keccak(text=sink) for sink in sinks]
+        self._parsers: List[LogParser] = _construct_parsers(self._w3, sinks)
+
+    def _receive_message(self) -> Optional[LogReceipt]:
```

Comment:
Let's just rework `BaseMessageProvider` that it can be used here without any queue.
As example, we can rework ` _receive_message` that it should return generator or iterable.

Reply:
Done